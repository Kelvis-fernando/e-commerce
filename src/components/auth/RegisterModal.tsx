import { LoginModalContainer } from "../../styles/auth/loginModal";
import TitleAndSubtitle from "../layout/TitleAndSubtitle";

const RegisterModal = () => {
  return (
    <LoginModalContainer>
      <TitleAndSubtitle
        title="Ainda nÃ£o tem uma conta?"
        subtitle="Adicione os seus dados para poder fazer os seus pedidos"
      />
      <form>
        <label htmlFor="name">Nome</label>
        <input type="text" name="name" id="name" />
        <label htmlFor="email">E-mail</label>
        <input type="email" name="email" id="email" />
        <label htmlFor="password">Senha</label>
        <input type="password" name="password" id="password" />
        <label htmlFor="password2">Confirme a senha</label>
        <input type="password" name="password2" id="password2" />
        <label htmlFor="userType">TIpo de usuario</label>
        <select name="userType" id="userType">
          <option value="">Selecione um tipo de usuario</option>
          <option value="user">Consumidor</option>
          <option value="producer">Produtor</option>
        </select>
        <button type="submit">Entrar</button>
      </form>
      <h2>Voltar</h2>
    </LoginModalContainer>
  );
};

export default RegisterModal;
