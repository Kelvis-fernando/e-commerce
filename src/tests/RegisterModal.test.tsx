import React from "react";
import { render } from "@testing-library/react";
import RegisterModal from "../components/auth/RegisterModal";

describe("RegisterModal", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<RegisterModal />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
