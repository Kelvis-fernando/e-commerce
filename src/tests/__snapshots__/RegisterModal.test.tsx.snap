// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterModal should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hlQNyq"
  >
    <div
      class="sc-dkrFOg ffpTbI"
    >
      <h1>
        Ainda n√£o tem uma conta?
      </h1>
      <h3>
        Adicione os seus dados.
      </h3>
    </div>
    <div
      class="sc-gswNZR cWphtV"
    >
      <form>
        <label
          for="name"
        >
          Nome
        </label>
        <input
          id="name"
          name="name"
          required=""
          type="text"
        />
        <label
          for="email"
        >
          E-mail
        </label>
        <input
          id="email"
          name="email"
          required=""
          type="email"
        />
        <label
          for="password"
        >
          Senha
        </label>
        <input
          id="password"
          name="password"
          required=""
          type="password"
        />
        <label
          for="confirmPassword"
        >
          Confirme a senha
        </label>
        <input
          id="confirmPassword"
          name="confirmPassword"
          required=""
          type="password"
        />
        <label
          for="typeOfUser"
        >
          Tipo de usuario
        </label>
        <select
          id="typeOfUser"
          name="typeOfUser"
        >
          <option
            value="user"
          >
            Consumidor
          </option>
          <option
            value="producer"
          >
            Produtor
          </option>
        </select>
        <button
          type="submit"
        >
          Cadastrar
        </button>
      </form>
      <h2>
        Voltar
      </h2>
    </div>
  </div>
</div>
`;
